CMakeLists.txt

```diff
@@ -0,0 +1,12 @@
+cmake_minimum_required(VERSION 3.15...3.30)
```

Comment:
Why do you set cmake max version?

Reply:
It's the range of versions advised in modern cmake. Cmake unfortunately likes to add breaking changes so it ensures that this will always work. 
https://cliutils.gitlab.io/modern-cmake/chapters/intro/dodonot.html